cloudferry_devlab/cloudferry_devlab/generate_load.py

```diff
@@ -239,51 +240,74 @@ def chech_image_state_on_dst(self, img_id):
                         self.config.images_not_included_in_filter]
+        default_filter_dict = {
+            'tenants': {
+                'tenant_id': []
+            },
+            'instances': {
+                'id': []
+            },
+            'images': {
+                'images_list': [],
+                'exclude_images_list': [],
+                'dont_include_public_and_members_from_other_tenants': False
+            }
+        }
+        filter_dict = copy.deepcopy(default_filter_dict)
+        file_path = {}
         for tenant in self.config.tenants + [{'name': self.tenant}]:
             if tenant.get('deleted'):
                 continue
-            filter_dict = {
-                'tenants': {
-                    'tenant_id': []
-                },
-                'instances': {
-                    'id': []
-                },
-                'images': {
-                    'images_list': [],
-                    'dont_include_public_and_members_from_other_tenants': False
-                }
-            }
-            filter_dict['tenants']['tenant_id'] = [str(
-                self.get_tenant_id(tenant['name']))]
-            self.switch_user(user=self.username, password=self.password,
-                             tenant=tenant['name'])
+            filter_dict, img_list, vm_list = self.get_tenant_resources(
+                tenant, filter_dict)
+            filter_dict = self.set_tenant_filter(tenant, filter_dict,
+                                                 img_list, vm_list,
+                                                 not_incl_img,
+                                                 all_tenants_filter)
+            if not all_tenants_filter:
+                file_path[self.config.filters_file_naming_template.format(
+                    tenant_name=tenant['name'])] = filter_dict
+                filter_dict = copy.deepcopy(default_filter_dict)
+            else:
+                file_path[self.config.all_tenants_filter_filename] = \
+                    filter_dict
 
-            vm_id_list = [x.__dict__['id'] for x in
-                          self.novaclient.servers.list()]
-
-            img_id_list = [x.__dict__['id'] for x in
-                           self.glanceclient.images.list(search_opts={
-                               'is_public': False})]
-
-            for key in filter_dict.keys():
-                if key == 'images':
-                    for img_id in img_id_list:
-                        if img_id not in not_incl_img:
-                            filter_dict[key]['images_list'].append(str(img_id))
-                elif key == 'instances':
-                    for vm in vm_id_list:
-                        if vm != self.get_vm_id('not_in_filter'):
-                            filter_dict[key]['id'].append(str(vm))
+        for key, value in file_path.iteritems():
+            with open(self.get_abs_path(key), "w") as f:
+                yaml.dump(value, f, default_flow_style=False)
```

Comment:
We'd need a more generic serialization engine to be used everywhere in the tests, this solution is not reusable. Please rework.